packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -99,124 +107,330 @@ const defaultProps = {
+    placeholder,
+    required,
+    selectedDate,
+    value,
+    ...customProps
+  } = props;
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const localeFormat = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateValue = DateUtil.convertToISO8601(value, DateUtil.getFormatByLocale(intl.locale));
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const placeholderValues = DateUtil.getPlaceholderValues(localeFormat, placeholder);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
+    }
+
+    // Clears the inputs when appropriate (e.g. excluded, out of range)
+    if (!selectedDate) {
+      dateDispatch({ day: '', month: '', year: '' });
     }
-  }
+  }, [selectedDate, shouldShowPicker, onClick]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, dateFormat)) {
+      const parseDate = {
+        dayValue: dateValue.substring(8, 10),
+        monthValue: dateValue.substring(5, 7),
+        yearValue: dateValue.substring(0, 4),
+      };
+
+      if (parseDate.yearValue && Object.values(parseDate).join('').length === 8) {
+        const formatISO = DateUtil.formatISODate(
+          parseDate.yearValue.concat('-',
+            parseDate.monthValue).concat('-',
+            parseDate.dayValue),
+          DateUtil.ISO_EXTENDED_DATE_FORMAT,
+        );
+        const formattedDate = DateUtil.strictFormatISODate(formatISO, dateFormat);
+        onChange({ target: { value: formattedDate }, isDefaultPrevented: () => {}, type: 'change' });
+        dateDispatch({
+          day: parseDate.dayValue,
+          month: parseDate.monthValue,
+          year: parseDate.yearValue,
+        });
+      }
+    }
+  }, [dateFormat, dateValue, onChange, value]);
 
-    if (!attributes.readOnly && this.onCalendarButtonClick && this.props.onClick) {
-      this.onCalendarButtonClick(event, this.props.onClick);
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {int} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateVariant - Date variant based on the currently supported date
```

Comment:
@benbcai may be able to provide more insights. I think you just updated some logic based on my previous comments. I was confused because I didn't realize you were trying to standardize the formatting order for parsing & composing the date value from the three inputs.

What are your thoughts on updating some of the naming to align with the concept naming associated with this?

I'm thinking `dateOrder` might be the clearest.

The search google returns https://en.wikipedia.org/wiki/Date_format_by_country which states:
> Order of the basic components [components of calendar date]:
.
    B – big-endian (year, month, day), e.g. 2006-04-22 or 2006.04.22 or 2006/04/22 or 2006 April 22
    L – little-endian (day, month, year), e.g. 22.04.2006 or 22/4/2006 or 22-04-2006 or 22 April 2006
    M – middle-endian (month, day, year), e.g. 04/22/2006 or April 22, 2006 

Then according to this doc: https://en.wikipedia.org/wiki/Calendar_date, there are four potential formats:
- DMY
- YMD
- MDY
- YMD
- YDM

Reply:
Looks like some places in DateUtil calls it `order` while other places have it `variant` so I think it would be easier to follow when they are more consistent. `dateOrder` or dateFormatOrder` would make sense. Among the locales we currently support, we would have 3 format orders - `DMY`, `YMD`, and `MDY`. We currently don't have a locale that uses the `YDM` order.